src/OSPSuite.Core/Domain/Formulas/DistributionFormulaFactory.cs

```diff
@@ -21,68 +25,85 @@ public interface IDistributionFormulaFactory
          IParameter minParameter,
          IParameter maxParameter);
 
+      DistributionFormula CreateFor(DistributionType distributionType, IDimension dimension);
    }
 
    public class DistributionFormulaFactory : IDistributionFormulaFactory
    {
       private readonly IObjectPathFactory _objectPathFactory;
       private readonly IObjectBaseFactory _objectBaseFactory;
 
-      public DistributionFormulaFactory(IObjectPathFactory objectPathFactory,IObjectBaseFactory objectBaseFactory)
+      public DistributionFormulaFactory(IObjectPathFactory objectPathFactory, IObjectBaseFactory objectBaseFactory)
       {
          _objectPathFactory = objectPathFactory;
          _objectBaseFactory = objectBaseFactory;
       }
 
-      public NormalDistributionFormula CreateNormalDistributionFormulaFor(IDistributedParameter distributedParameter, IParameter meanParameter, IParameter deviationParameter)
+      public NormalDistributionFormula CreateNormalDistributionFormulaFor(IDistributedParameter distributedParameter, IParameter meanParameter, IParameter deviationParameter) =>
```

Comment:
those changes are pretty cool here. I suggest looking at them side by side to understand how i have use a func that returns a func to minimize the parameters that I had to pass around. Functional programming for the win

Reply:
nice :)